drivers/adc/adc_mcp356xr.c

```diff
@@ -0,0 +1,1317 @@
+
+#define ADC_MCP356XR_INIT_CONFIG3(index)                                                           \
+	MCP356XR_REGISTER_FIELD_PREP(CONFIG3, CONV_MODE,                                           \
+				     MCP356XR_REGISTER_CONFIG3_CONV_MODE_ONE_SHOT_GO_STANDBY) |    \
+		MCP356XR_REGISTER_FIELD_PREP(                                                      \
+			CONFIG3, DATA_FORMAT,                                                      \
+			MCP356XR_REGISTER_CONFIG3_DATA_FORMAT_8_BITS_SIGN_24_BITS_DATA) |          \
+		MCP356XR_REGISTER_FIELD_PREP(CONFIG3, CRC_FORMAT,                                  \
+					     MCP356XR_REGISTER_CONFIG3_CRC_FORMAT_16_BITS_CRC) |   \
+		MCP356XR_REGISTER_FIELD_PREP(CONFIG3, EN_CRCCOM,                                   \
+					     ADC_MCP356XR_INIT_CONFIG3_EN_CRCCOM) |                \
+		MCP356XR_REGISTER_FIELD_PREP(CONFIG3, EN_OFFCAL,                                   \
+					     MCP356XR_REGISTER_CONFIG_FLAG_DISABLED) |             \
+		MCP356XR_REGISTER_FIELD_PREP(CONFIG3, EN_GAINCAL,                                  \
+					     MCP356XR_REGISTER_CONFIG_FLAG_DISABLED)
+
+#define ADC_MCP356XR_INIT_IRQ(index)                                                               \
+	MCP356XR_REGISTER_FIELD_PREP(IRQ, EN_FASTCMD, MCP356XR_REGISTER_CONFIG_FLAG_ENABLED) |     \
+		MCP356XR_REGISTER_FIELD_PREP(IRQ, EN_STP,                                          \
+					     MCP356XR_REGISTER_CONFIG_FLAG_DISABLED) |             \
+		MCP356XR_REGISTER_FIELD_PREP(IRQ, IRQ_MODE, DT_INST_IRQ_MODE(index))
+
+#define ADC_MCP356XR_INIT(index)                                                                   \
+	static const struct adc_mcp356xr_config adc_mcp356xr_config_##index = {                    \
+		.spi = SPI_DT_SPEC_INST_GET(index, MCP356XR_SPI_OPERATION, MCP356XR_SPI_CS_DELAY), \
+		.address = DT_INST_PROP(index, address),                                           \
+		.vref_settle_time = K_MSEC(DT_INST_PROP_OR(index, vref_settle_time_ms, 0)),        \
+		.init_reg_value =                                                                  \
+			{                                                                          \
+				[REGISTER_INDEX(CONFIG0)] = ADC_MCP356XR_INIT_CONFIG0(index),      \
+				[REGISTER_INDEX(CONFIG1)] = ADC_MCP356XR_INIT_CONFIG1(index),      \
+				[REGISTER_INDEX(CONFIG2)] = ADC_MCP356XR_INIT_CONFIG2(index),      \
+				[REGISTER_INDEX(CONFIG3)] = ADC_MCP356XR_INIT_CONFIG3(index),      \
+				[REGISTER_INDEX(IRQ)] = ADC_MCP356XR_INIT_IRQ(index),              \
+				[REGISTER_INDEX(MUX)] = 0,                                         \
+			},                                                                         \
+		IF_DISABLED(CONFIG_ADC_MCP356XR_POLL,                                              \
+			    (.irq = GPIO_DT_SPEC_INST_GET(index, irq_gpios),))                     \
+	};                                                                                         \
```

Comment:
clang-format produces strange result on this (curly brace is placed at the end of the previous line, which is curious)